lib/fog/aws/models/storage/directory.rb

```diff
@@ -30,6 +37,38 @@ def destroy
           false
         end
 
+        # @param options [Hash] (defaults to: {}) — a customizable set of options
+        # @option options max_attempts [Integer] — default: 3 — Maximum number of
+        #   times to attempt to delete the empty bucket.
+        # @option options initial_wait [Float] — default: 1.3 — Seconds to wait before
+        #   retrying the call to delete the bucket, exponentially increased for each attempt.
+        def destroy!(options = {})
+          requires :key
+          options = {
```

Comment:
I think for consistency it might make sense to convert this to use `Fog.wait_for { ... }`? It has defaults for intervals/back-off already built in and is what other things use. It would change from a number of max_attempts to a timeout though, if I recall correctly. I just thought it might make it easier to understand/adopt/maintain to reuse the existing mechanism if we think it is similar enough. What do you think? The implementation is here: https://github.com/fog/fog-core/blob/7865ef77ea990fd0d085e49c28e15957b7ce0d2b/lib/fog/core/wait_for.rb and spec here: https://github.com/fog/fog-core/blob/7865ef77ea990fd0d085e49c28e15957b7ce0d2b/spec/wait_for_spec.rb

Reply:
looks good, will use that instead then